File: tojson.js
var yaml = require('yamljs');
var fs = require('fs');
var refParser = require('json-schema-ref-parser');

var ymlFile;
fs.readFile('swagger.yml', 'utf8', function(err, data) {
  if (err) throw err;
  var json = yaml.parse(data);
  refParser.dereference(json, function(err, schema) {
    if (err) {
      console.error(err);
    } else {
      fs.writeFile('swagger.json', JSON.stringify(schema, null, 2).concat('\n'), function(err) {
        if (err) return console.log(err);
        console.log('yml converted to json');
      });
    }
  });
});

File: to_json.rb
require 'json'
require 'json_schema_ref_parser'

def read_yml_file(file)
  #yml_file = File.open(file, 'r')
  #yml = yml_file.read
  #yml_file.close
  #YAML::load(yml)
  YAML.load_file(file)
end

def write_json_file(json)
  File.open('openapi.json', 'w') { |file| file.write(json) }
end

schema = read_yml_file('openapi.yml')
result = JsonSchemaRefParser::RefParser.dereference(schema)
write_json_file(JSON.pretty_generate(result))
